testsuite/gateway/gateway_api/gateway.py

```diff
@@ -125,7 +125,7 @@ def get_tls_cert(self) -> Optional[Certificate]:
         tls_cert = Certificate(
             key=tls_cert_secret["tls.key"],
             certificate=tls_cert_secret["tls.crt"],
-            chain=tls_cert_secret["ca.crt"],
+            chain=tls_cert_secret["ca.crt"] if "ca.crt" in tls_cert_secret else None,
```

Comment:
iirc we were setting the certificate also as a chain if it didn't have one. I think we can do the same here and later, for the test, connect the root certificate with the certificate chain instead
```suggestion
            chain=tls_cert_secret["ca.crt"] if "ca.crt" in tls_cert_secret else tls_cert_secret["tls.crt"],
```

Reply:
While not neccesarily bad idea, I would do it in the subsequent PR (we need to redo all of our certificates and tests) and let this merge, since it is ok for the usecase it is needes

Reply:
ack 👍. Also, a Certificate dataclass doesn't show that it supports any Optional values, but I guess it doesn't complain about it